school_center.py

```diff
@@ -70,10 +70,19 @@ def sort_key(c):
 
 def read_tsv(file_path: str) -> List[Dict[str, str]]:
     data = []
-    with open(file_path, 'r', newline='', encoding='utf-8') as file:
-        reader = csv.DictReader(file, delimiter='\t')
-        for row in reader:
-            data.append(dict(row))
+    try:
+        with open(file_path, 'r', newline='', encoding='utf-8') as file:
+            reader = csv.DictReader(file, delimiter='\t')
+            for row in reader:
+                data.append(dict(row))
+    except FileNotFoundError:
+        print(f"Error: File '{file_path}' not found.")
+    except PermissionError:
+        print(f"Error: Permission denied while accessing file '{file_path}'.")
+    except IOError:
+        print(f"Error opening or reading file: {file_path}")
+    except Exception as e:
+        print(f"Error: An unexpected error occurred while reading file '{file_path}': {e}")
```

Comment:
A few things to note: 

* Let's add a parameterized test to ensure that these exceptions are caught properly. 
* Also, it might be better to use a logger than a print statement. 
* Returning empty data might be misleading when an exception occurs because we are returning empty data regardless. Ideally, it should exit as soon as an exception occurs and the user should be prompted accordingly.

Reply:
good insights! I will try to implement those